Add ST X-NUCLEO-EEPRMA2 EEPROM shield and a EEPROM driver sample

str4t0m opened the pull request:
This PR adds the ST X-NUCLEO-EEPRMA2 as a new Arduino shield and adds a minimal sample to demonstrate the EEPROM driver api.

str4t0m committed:
shields: add st x-nucleo-eeprma2 eeprom shiled for m24xx, m95xx devices

This commit adds st microelectronics x-nucleo-eeprma2 shield,
which has populated 6 different i2c and spi eeproms.
They are all compatible with the existing at24 and at25 drivers.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m committed:
samples: add eeprom sample

This commit adds a simple application to demonstrate the zephyr
eeprom interface.
It uses an eeprom to keep track of the number reboots.

Signed-off-by: Thomas Stranger <thomas.stranger@outlook.com>

str4t0m commented:
I see from @henrikbrixandersen as eeprom maintainer no review was requested.
Could you have a look at this?

erwango commented on samples/drivers/eeprom/sample.yaml:
```diff
@@ -0,0 +1,16 @@
+        - "Reset the MCU to see the inceasing boot counter."
+
+tests:
+  sample.drivers.eeprom:
+    tags: eeprom
+    platform_allow: native_posix
```
Adding a variant with shield would be nice. Not blocking

str4t0m replied:
Added a configuration, but I am not sure about the integration platforms.
I think most boards are already filtered due to console harness?

str4t0m replied:
Only native_posix is now listed as integration platform.

erwango reviewed (approved):
Looks OK from the shield point of view.
Need EEPROM maintainer to confirm for sample

str4t0m commented:
I have added a "#include <freq.h>" in the shield overlay as most platforms with the exception of ST haven't included this themself.
And added a test configuration for the shield in the eeprom sample.

henrikbrixandersen reviewed (changes requested):
A couple of minor nits, otherwise looks great.

str4t0m commented on samples/drivers/eeprom/README.rst:
```diff
@@ -0,0 +1,48 @@
+   :shield: x_nucleo_eeprma2
+   :compact:
+
+
```
Done

henrikbrixandersen commented:
@str4t0m If you could you help verify #35743 on this shield I would be grateful.

str4t0m commented:
The eeprom tests still pass on the shield.

I wonder whether I should add an eeprom-1 alias for the eeprom4 node to the shield, such that one I2C and one SPI device are active?

henrikbrixandersen commented:
Good idea. Also, thanks for testing!
